extensions/fetch/lib.rs

```diff
@@ -568,12 +601,19 @@ pub fn create_http_client(
   ca_data: Option<Vec<u8>>,
   proxy: Option<Proxy>,
 ) -> Result<Client, AnyError> {
+  let mut tls_config = ClientConfig::new();
+  tls_config.set_persistence(CLIENT_SESSION_MEMORY_CACHE.clone());
+  tls_config.root_store = load_native_certs()
+    .expect("could not load platform certs");
+  tls_config.root_store
+    .add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
```

Comment:
I can confirm that this successfully loads my companies CA stored in my system keychain and lets me access internal domains.

It should also pull in the default webpki certs, so if those certs go stale over time a user would be able to either upgrade their version of Deno or install the missing certs into their keychain.

This should cover a broad range of scenarios by default though.